### PR TITLE
Fix Snap Server Hive Tests

### PR DESCRIPTION
**Pull request by asdacap**
- Fix snap server hive tests.
- Need to change the default config to turn on snap serving and snap sync, but otherwise should pass all snap tests.
- Slight performance improvement. Halfpath peaks at 250k nodes per sec now.

#### What types of changes does your code introduce?

- [X] Bugfix (a non-breaking change that fixes an issue)

#### Requires testing

- [X] Yes
- [ ] No

#### If yes, did you write tests?

- [X] Yes
- [ ] No

#### Notes on testing

- Have not test sync yet.

### CODE REVIEW

**Review by benaadams — APPROVED** [review]

File: src/Nethermind/Nethermind.Network.Test/P2P/Subprotocols/Snap/Messages/TrieNodesMessageSerializerTests.cs [hunk 1]
```diff
@@ -20,5 +22,16 @@ public void Roundtrip()
 
             SerializerTester.TestZero(serializer, message);
         }
+
+        [Test]
+        public void RoundtripWithCorrectLength()
+        {
+            byte[][] data = { new byte[] { 0xde, 0xad, 0xc0, 0xde }, new byte[] { 0xfe, 0xed } };
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network.Test/P2P/Subprotocols/Snap/Messages/TrieNodesMessageSerializerTests.cs:29]: minor: [] are sexier now ;)

File: src/Nethermind/Nethermind.Blockchain/Utils/LastNStateRootTracker.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                    (_,
+                        oldValue) => oldValue - 1);
+                if (newNum == 0) _availableStateRoots.Remove(oldStateRoot, out _);
+            }
+            _stateRootQueue.Enqueue(newHead.StateRoot);
+            _lastQueuedStateRoot = newHead.StateRoot;
+            return;
+        }
+
+        List<Hash256> stateRoots = new();
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Blockchain/Utils/LastNStateRootTracker.cs:66]: ArrayPoolList?

File: src/Nethermind/Nethermind.Synchronization/SnapSync/SnapServer.cs [hunk 3]
```diff
@@ -47,45 +47,24 @@ public class SnapServer : ISnapServer
...
-            _rootWithMissingNode.Delete(rootHash);
-        }
-
-        return false;
-    }
-
-    private void TrackTrieNodeException(in ValueHash256 rootHash)
-    {
-        _rootWithMissingNode.Set(rootHash, DateTimeOffset.Now);
+        return !_stateRootTracker.HasStateRoot(stateRoot.ToCommitment());
```
[the remark this answers is not in the document]
Comment (asdacap) [reply src/Nethermind/Nethermind.Synchronization/SnapSync/SnapServer.cs:67]: messages?

**Comment by asdacap** [comment]
Yes we are passing all test. Probably need to modify them so that snap serving is enabled but thats it.

**Pull request merged**